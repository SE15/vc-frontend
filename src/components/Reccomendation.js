import {
    Button, Modal, ModalOverlay, ModalContent
    , ModalCloseButton, Text, useDisclosure, ModalFooter, Textarea
    , Avatar, Stack, Heading, Box
} from '@chakra-ui/react';
const Reccomendation = () => {
    const { isOpen, onOpen, onClose } = useDisclosure()

    return (
        <>
            <Button onClick={onOpen} bg="blueGreen.200">Post Reccomendation</Button>
            <Modal isOpen={isOpen} onClose={onClose} color="black.700" closeOnOverlayClick="false">
                <ModalOverlay />
                <ModalContent>
                    <Box bg="white.200" w="100%" p={4} color="white">
                        <stack>
                            <Heading color="black.400" fontSize="lg">
                                Post Reccomendation
                </Heading>
                            <br />
                            <Stack direction="row">
                                <Avatar name="Oshigaki Kisame" src="https://bit.ly/broken-link" />

                                <Text color="blue.100">
                                    Oshisidj Kisksls
            </Text>

                                <Textarea placeholder="Enter reccomendation" color="black.400" />
                                <ModalCloseButton />
                            </Stack>
                        </stack>
                        <ModalFooter>
                            <Button colorScheme="blue" mr={3} >
                                Post
            </Button>
                        </ModalFooter>
                    </Box>
                </ModalContent>

            </Modal>

        </>
    )
}
export default Reccomendation;