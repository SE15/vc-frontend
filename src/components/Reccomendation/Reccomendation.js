import { React } from "react"

import {Flex,Image, Text,Avatar, Stack, Heading, Box, Divider
} from '@chakra-ui/react';

const Reccomendation = props => {
    return (
        <>
        <Flex>
            <Box bg="white.200" w="50%" p={4} color="white">
                <Stack>  
                    <Stack direction="row">
                        if({props.postedImage}!=null){
                            <Image src={props.postedImage} size="100%" rounded="1rem" shadow="2xl" />
                        }
                        else{
                            <Avatar name={props.postedBy} src="https://bit.ly/broken-link" />
                        }
                        <Heading color="black.400" fontSize="lg">
                            {props.postedBy}
                        </Heading>
        
                    </Stack>
                    <Divider/>
                    <Text color="black.900" fontSize="md" textAlign="left">
                        {props.reccomendation}
                    </Text>
                </Stack>
            </Box>
            </Flex>
        </>
    )
}
export default Reccomendation