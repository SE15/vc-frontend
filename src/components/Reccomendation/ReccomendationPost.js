import  React, {useState} from "react"

import PropTypes from "prop-types"
import {
    Button, Modal, ModalOverlay, ModalContent, Image
    , ModalCloseButton, Text, useDisclosure, ModalFooter,
     Textarea, Avatar, Stack, Heading, Box
} from '@chakra-ui/react';
import { kPrimaryBlackLight,kSecondaryBlueLight } from '../../constants';
import { Component } from "react";
import Reccomendation from "./Reccomendation";

class ReccomendationPost extends Component{
    constructor(props){
        super(props);    
    }
    render(){
    return (
        <>
            <ReccomendationButton id="rec_button"visit={props.visit} name={props.name} />
        </>
    );
    }
}



class ReccomendationButton extends Component{
    constructor(props){
        super(props);
        this.state = {
            isLoggedIn: true,
            isOpen: false,
            isLoading: false,
            isreccomended: false,
            error: '',
            
        };
    }
    onClose=()=>{
        this.setState({
            ...this.state,
            isOpen: false
        }); 
    }
    onOpen= ()=>{
        this.setState({
            ...this.state,
            isOpen: true
        }); 
    }


    handleReccomendation= ()=>{
        this.setState({
            ...this.state,
            isLoading: true
        });
        try{
            //wait for update validation in db

            this.setState({
                ...this.state,
                isreccomended: true,
                isLoading: false
            });
        

        }catch{
            this.setState({
                ...this.state,
                isreccomended: false,
                isLoading: false,
                error: "Couldn't Post Reccomendation"
            });

        }
    }
    render(){
    if (this.props.visit == true && this.state.isLoggedIn == true && this.state.isreccomended==false) {

        return (
            <>
                <Box>
                    <Button id="post" onClick={()=>this.onOpen()} bg={kSecondaryBlueLight} isDisabled={false}>Post Reccomendation</Button>
                </Box>
                <Modal isOpen={this.state.isOpen} onClose={()=>onClose()} color="black.700" closeOnOverlayClick="false">
                    <ModalOverlay />
                    <ModalContent>
                        <Box bg={kPrimaryBlackLight} w="100%" p={4} color="white">
                            <Stack>
                                <Heading color="black.400" fontSize="lg">
                                    Post Reccomendation
                                </Heading>
                                <br />
                                <Stack direction="row">
                                    {this.props.cimage!=null?(
                                        <Image src={this.props.image}  borderRadius="full" boxSize="3rem" />
                                    ) :(
                                        <Avatar name={this.props.name} src="https://bit.ly/broken-link" />
                                    )
                                    }
                                    <Text color={kSecondaryBlueLight}>
                                        {this.props.name}
                                    </Text>
                   
                                    <Textarea placeholder="Enter reccomendation" color="black.400" />
                                    
                                    <ModalCloseButton />
                                </Stack>
                            </Stack>
                            <ModalFooter>
                                <Button colorScheme="blue" mr={3} onClick={()=>handleReccomendation()}>
                                    Post
                                </Button>
                            </ModalFooter>
                        </Box>
                    </ModalContent>
                </Modal>
            </>
        );
    } else {
        return (
            <Button onClick={()=>onOpen()} bg={kSecondaryBlueLight}isDisabled={true}>Post Reccomendation</Button>
        );
    }
    }   
}

export {
    ReccomendationPost,
    ReccomendationButton
} 