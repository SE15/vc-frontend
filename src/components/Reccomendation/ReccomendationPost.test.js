import React,{useState} from 'react';

import {configure,mount,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'


import {ReccomendationButton,ReccomendationPost} from './ReccomendationPost'
import { Button, Modal } from '@chakra-ui/react';
import Reccomendation from './Reccomendation';

configure({adapter:new Adapter()});

describe('<ReccomendationPost />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper =shallow(<ReccomendationButton name="Anne De Mel"/>);
    });


    it('should set isreccomended true when reccomendation is posted',()=>{
        wrapper.setProps({visit: true});
        expect(wrapper.instance().state.isreccomended).toBe(false);
        wrapper.instance().handleReccomendation()
        expect(wrapper.instance().state.isreccomended).toBe(true);
    });

    it('should disable post reccomend button when not logged in',()=>{
        wrapper.setState({isLoggedIn: false});
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should disable post reccomend button when reccomendation is added',()=>{
        wrapper.setProps({visit: true});
        wrapper.instance().handleReccomendation();
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should disable post reccomend button when user viewing own profile',()=>{
        wrapper.setProps({visit: false});
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should display post reccomend button if its a profile visit & reccomendation not added before',()=>{
        wrapper.setProps({visit:true});
        wrapper.setState({isLoggedIn: true});
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should contain a modal',()=>{
        wrapper.setProps({visit:true});
        wrapper.setState({isLoggedIn: true});
        expect(wrapper.find(Modal)).toHaveLength(1);
    });

    it('should open a modal when button clicked',()=>{
        wrapper.setProps({visit:true});
        wrapper.setState({isLoggedIn: true});
        const wrap=wrapper.find(Button).at(0);
        //to implement
    });



});    