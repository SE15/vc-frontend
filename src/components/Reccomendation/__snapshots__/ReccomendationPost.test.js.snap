// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReccomendationPost /> should disable post reccomend button when not logged in 1`] = `
Array [
  <Button
    bg="#4f83cc"
    isDisabled={true}
    onClick={[Function]}
  >
    Post Reccomendation
  </Button>,
]
`;

exports[`<ReccomendationPost /> should disable post reccomend button when reccomendation is added 1`] = `
Array [
  <Button
    bg="#4f83cc"
    isDisabled={true}
    onClick={[Function]}
  >
    Post Reccomendation
  </Button>,
]
`;

exports[`<ReccomendationPost /> should disable post reccomend button when user viewing own profile 1`] = `
Array [
  <Button
    bg="#4f83cc"
    isDisabled={true}
    onClick={[Function]}
  >
    Post Reccomendation
  </Button>,
]
`;

exports[`<ReccomendationPost /> should display post reccomend button if its a profile visit & reccomendation not added before 1`] = `
Array [
  <React.Fragment>
    <Box>
      <Button
        bg="#4f83cc"
        id="post"
        isDisabled={false}
        onClick={[Function]}
      >
        Post Reccomendation
      </Button>
    </Box>
    <Modal
      allowPinchZoom={false}
      autoFocus={true}
      blockScrollOnMount={true}
      closeOnOverlayClick="false"
      color="black.700"
      isOpen={false}
      lockFocusAcrossFrames={true}
      motionPreset="scale"
      onClose={[Function]}
      returnFocusOnClose={true}
      scrollBehavior="outside"
      trapFocus={true}
    >
      <ModalOverlay />
      <ModalContent>
        <Box
          bg="#4f5b62"
          color="white"
          p={4}
          w="100%"
        >
          <Stack>
            <Heading
              color="black.400"
              fontSize="lg"
            >
              Post Reccomendation
            </Heading>
            <br />
            <Stack
              direction="row"
            >
              <Avatar
                name="Anne De Mel"
                src="https://bit.ly/broken-link"
              />
              <Text
                color="#4f83cc"
              >
                Anne De Mel
              </Text>
              <Textarea
                color="black.400"
                placeholder="Enter reccomendation"
              />
              <ModalCloseButton />
            </Stack>
          </Stack>
          <ModalFooter>
            <Button
              colorScheme="blue"
              mr={3}
              onClick={[Function]}
            >
              Post
            </Button>
          </ModalFooter>
        </Box>
      </ModalContent>
    </Modal>
  </React.Fragment>,
]
`;
